File: core/tests/fixtures/mothers/book_sale_mother.py
from src.domain.book import BookSale
from tests.fixtures.mothers.base import BaseMother


class BookSaleMother(BaseMother):
    """Mother class for creating BookSale instances for testing purposes."""

    _VALID = {
        'saleable': True,
        'price': 100.0,
        'discount': 10.0,
    }

    @classmethod
    def valid(cls, **overrides) -> BookSale:
        """Returns a valid BookSale instance."""
        return cls._create(cls._VALID, overrides)

    @classmethod
    def _create(cls, values: dict, overrides: dict) -> BookSale:
        """Creates a BookSale instance with the given overrides."""
        data = cls._merge(values, overrides)
        return BookSale(**data)

File: editor/tests/fixtures/mothers/book-sale.mother.ts
import { BaseMother } from '@tests/fixtures/mothers/base/base-mother';
import { BookSale } from '@/books/model/book-sale';

const VALID = {
  saleable: true,
  price: 100.0,
  discount: 10.0,
};

type BookSaleFixture = typeof VALID;

export class BookSaleMother extends BaseMother {
  static valid(this: void, overrides?: Partial<BookSale>): BookSale {
    return BookSaleMother.create(VALID, overrides);
  }

  protected static create(
    values: BookSaleFixture,
    overrides?: Partial<BookSale>,
  ): BookSale {
    const fields: BookSale = BookSaleMother.merge<BookSaleFixture, BookSale>(
      values,
      overrides,
    );

    return BookSale.from(fields);
  }
}
